code/modules/projectiles/projectile/bullets/shotgun.dm

```diff
@@ -96,6 +96,18 @@
 	stamina = 6
 	embedding = null
 
+/obj/projectile/bullet/pellet/shotgun_penetrator
+	name = "penetrator pellet"
+	damage = 8
+	embedding = null
+	armour_penetration = 20 //P E N E T R A T O R
+	projectile_piercing = PASSMOB
+
+/obj/projectile/bullet/pellet/shotgun_penetrator/on_hit(atom/target)
+	. = ..()
+	if(iscarbon(target))
+		projectile_piercing = ~PASSMOB // The pellets penetrate ONE person!
```

Comment:
I am thinking that you should move this to prehit_pierce, also use pass_flags_self instead of iscarbon (target.pass_flags_self & PASSMOB).

Reply:
prehit_pierce() just causes the ~PASSMOB to act before it even passes the mob itself, resulting in only a single mob hit. At least, that's what seems to be happening in-game.

Reply:
how are you doing it in prehit_pierce?
you want something like:
```
if (projectile_piercing & target.pass_flags_self & PASSMOB)
	projectile_piercing = ~PASSMOB // The pellets penetrate ONE person!
	return PROJECTILE_PIERCE_HIT
```